apps/survey/contracts/Survey.sol

```diff
@@ -0,0 +1,250 @@
+
+
+contract Survey is AragonApp {
+    using SafeMath for uint256;
+    using SafeMath64 for uint64;
+
+    uint256 constant public PCT_BASE = 10 ** 18; // 0% = 0; 1% = 10^16; 100% = 10^18
+    uint256 constant public NO_VOTE = 0;
+    bytes32 constant public CREATE_SURVEYS_ROLE = keccak256("CREATE_SURVEYS_ROLE");
+    bytes32 constant public MODIFY_PARTICIPATION_ROLE = keccak256("MODIFY_PARTICIPATION_ROLE");
+
+    MiniMeToken public token;
+    uint256 public minAcceptParticipationPct;
+    uint64 public surveyTime;
+
+    struct OptionCast {
+        uint256 optionId;
+        uint256 stake;
+    }
+
+    /** To allow multiple option votes.
+     * Index 0 will always be for NO_VOTE option
+     * option Ids must be in ascending order
+     */
+    struct MultiOptionVote {
+        uint256 optionsCastedLength;
+        mapping (uint256 => OptionCast) castedVotes;
+    }
+
+    struct SurveyStruct {
+        address creator;
+        uint64 startDate;
+        uint256 options;
+        uint256 snapshotBlock;
+        uint256 minAcceptParticipationPct;
+        uint256 votingPower;                    // total tokens that can cast a vote
+        uint256 participation;                  // tokens that casted a vote
+        string metadata;
+
+        mapping (uint256 => uint256) votes;     // option -> voting power for option
+        mapping (address => MultiOptionVote) voters;    // voter -> options voted, with its stakes
+    }
+
+    SurveyStruct[] surveys;
+
+    event StartSurvey(uint256 indexed surveyId);
+    event CastVote(uint256 indexed surveyId, address indexed voter, uint256 option, uint256 stake);
+    event ChangeMinParticipation(uint256 minAcceptParticipationPct);
+
+    /**
+    * @notice Initializes Survey app with `_token.symbol(): string` for governance, minimum acceptance participation of `(_minAcceptParticipationPct - _minAcceptParticipationPct % 10^14) / 10^16` and durations of `(_surveyTime - _surveyTime % 86400) / 86400` day `_surveyTime >= 172800 ? 's' : ''`
+    * @param _token MiniMeToken address that will be used as governance token
+    * @param _minAcceptParticipationPct Percentage of total voting power that must participate in a survey for it to be taken into account (expressed as a 10^18 percentage, (eg 10^16 = 1%, 10^18 = 100%)
+    * @param _surveyTime Seconds that a survey will be open for token holders to vote
+    */
+    function initialize(
+        MiniMeToken _token,
+        uint256 _minAcceptParticipationPct,
+        uint64 _surveyTime
+    ) onlyInit external
+    {
+        initialized();
+
+        require(_minAcceptParticipationPct > 0 && _minAcceptParticipationPct <= PCT_BASE);
+
+        token = _token;
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+        surveyTime = _surveyTime;
+    }
+
+    /**
+    * @notice Change minimum acceptance participation to `(_minAcceptParticipationPct - _minAcceptParticipationPct % 10^14) / 10^16`%
```

Comment:
Well, I think the precedence is right, what I'm trying to do is remove decimals. So imagine that we have 66.66% and 100 instead of 10^14, and therefore `_minAcceptParticipationPct` is 6666. So we would do: 6666 - 6666 % 100 = 6666 - 66 = 6600. So when we divide by 100 it's exact. The error is that we have 10^14 and 10^16. Fixing it.

Reply:
Now I realize that my idea was to keep 2 decimals, that's why I had 10^14 and 10^16, but they were in reverse order.